webapp/views.py

```diff
@@ -453,21 +454,65 @@ def advantage_view():
 
             raise http_error
 
+        listings_response = advantage.get_marketplace_product_listings(
+            "canonical-ua"
+        )
+
+        product_listings = listings_response.get("productListings")
+
+        monthly_products = {}
+        yearly_products = {}
+        listings_to_products = {}
+
+        for listing in product_listings:
+            listings_to_products[listing["id"]] = listing["productID"]
+
+            if listing["period"] == "yearly":
+                yearly_products[listing["productID"]] = {
+                    "id": listing["id"],
+                    "metric": listing["allowanceMetric"],
+                    "price": listing["price"],
+                    "quantity": 0,
+                }
+
+            monthly_products[listing["productID"]] = {
+                "id": listing["id"],
+                "metric": listing["allowanceMetric"],
+                "price": listing["price"],
+                "quantity": 0,
+            }
```

Comment:
If the product is yearly, you will end up adding it to both `yearly_products` and to `monthly_products`.

Reply:
Fixed!